crates/sui-core/src/authority/authority_store.rs

```diff
@@ -823,6 +824,61 @@ impl AuthorityStore {
         Ok(())
     }
 
+    pub async fn bulk_insert_live_objects(
+        perpetual_db: &AuthorityPerpetualTables,
+        live_objects: impl Iterator<Item = LiveObject>,
+        indirect_objects_threshold: usize,
+        expected_sha3_digest: &[u8; 32],
+    ) -> SuiResult<()> {
+        let mut hasher = Sha3_256::default();
+        let mut batch = perpetual_db.objects.batch();
+        for object in live_objects {
+            hasher.update(object.object_reference().2.inner());
+            match object {
+                LiveObject::Normal(object) => {
+                    let StoreObjectPair(store_object_wrapper, indirect_object) =
+                        get_store_object_pair(object.clone(), indirect_objects_threshold);
+                    batch.insert_batch(
+                        &perpetual_db.objects,
+                        std::iter::once((
+                            ObjectKey::from(object.compute_object_reference()),
+                            store_object_wrapper,
+                        )),
+                    )?;
+                    if let Some(indirect_object) = indirect_object {
+                        batch.insert_batch(
```

Comment:
If there are two objects that share the same digest, `insert_batch` will result in an incorrect ref count value. 
FWIU snapshots shouldn't contain multiple versions of the same objects, making this edge case unlikely. 
So for simplicity you can use a full merge instead of an insert here. 
Although it is somewhat expensive, considering the rarity of the scenario, it is more compact than dealing with partial merges

Reply:
gotcha, sure I will make it a full merge.

Reply:
done